inbm-v5.md

```diff
@@ -0,0 +1,335 @@
+      - _Dynamic_: Information which constantly changes and is ideal to be plotted on a `time-sereis` database (e.g. CPU usage, memory usage etc)
+
+1. #### aggregator
+   - **Function**: Thread reporting status to `inbm-daemon`
+   - **Main Tasks**: This thread performs the task of combining and reporting information from platform to be sent to the cloud or `inbc` tool:
+      - _Event_: information that needs to be sent during the update process to indicat the progress
+      - _Status_: final result of an update
+      - _Telemetry_: `static` and `dynamic` telemetry
+
+Data Flow
+---------
+
+.. 
+   Guidelines:
+   1. Break down the system’s data flow into stages.
+   2. Highlight how components interact and communicate.
+
+Workflow Stages
+~~~~~~~~~~~~~~~
+
+1. **Stage 1: Input**:
+   - Example: Resource requests submitted via APIs or command-line tools.
+
+2. **Stage 2: Processing**:
+   - Example: Scheduling, allocation of resources, or preprocessing of data.
+
+3. **Stage 3: Output**:
+   - Example: Execution of workloads, generation of results, or API responses.
+
+Extensibility
+-------------
+
+.. 
+   Guidelines:
+   1. Explain how the system can be customized or extended.
+   2. Include modular integration points or plugin options.
+
+- **Adding Components**: [Explain how to add new modules or features].
+- **Customizing Workflows**: [Describe how to adapt the system to unique requirements].
+- **Integration Points**: [List APIs or interfaces for external integrations].
+
+Deployment
+----------
+
+[Content of Deployment]
+
+
+Technology Stack
+----------------
+
+Implementation
+~~~~~~~~~~~~~~
+
+.. 
+   Guidelines:
+   1. Information about how the app was implemented.
+
+[Content of Implementation]
+
+System Diagram
+~~~~~~~~~~~~~~
+
+.. 
+   Guidelines:
+   1. Include a diagram to illustrate how the system is deployed and what other applications it may be connected to.
+   2. Clearly label components, workflows, and integration points.
+
+.. figure:: ./images/stack-diagram.png
+   :alt: Technology Stack of [System or Tool Name]
+
+   Figure 1: Technology Stack of [System or Tool Name]
+
+Integrations
+~~~~~~~~~~~~
+
+.. 
+   Guidelines:
+   1. List the integrations between this application and other tech stack applications or systems.
+   2. Include links to additional material if available, including the development project, code, diagrams, and issues.
+
+[Content of Integrations]
+
+Security
+--------
+
+.. 
+   Guidelines:
+   1. Provide a brief overview of the security measures in place.
+   2. Explain the importance of security for the project.
+
+Security Policies
+~~~~~~~~~~~~~~~~~
+
+.. 
+   Guidelines:
+   1. Describe the security policies in place.
+   2. Include information on data protection, user privacy, and compliance.
+
+[Content of security policies]
+
+Authentication
+~~~~~~~~~~~~~~
+
```

Comment:
authentication for this design is simple: clouds and inbc users have identity `root` or `system administrator` and can execute any valid command/manifest